rmf_traffic/include/rmf_traffic/reservations/reservations.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef RMF_TRAFFIC__RESERVATIONS_HPP
+#define RMF_TRAFFIC__RESERVATIONS_HPP
```

Comment:
Let's name this header after the class that it's responsible for defining rather than just `reservations.hpp`. Alternatively we could consider just making the header `rmf_traffic/reservations.hpp` if we feel certain that the whole reservation system is best to hold entirely in one header.